vertx-sql-client/src/main/java/io/vertx/sqlclient/Pool.java

```diff
@@ -31,6 +37,63 @@
  */
 @VertxGen
 public interface Pool extends SqlClient {
+  
+  /**
+   * Create a connection pool to the database configured with the given {@code connectOptions} and default {@link PoolOptions}
+   *
+   * @param connectOptions the options used to create the connection pool, such as database hostname
+   * @return the connection pool
+   */
+  static Pool pool(SqlConnectOptions connectOptions) {
+    return pool(connectOptions, new PoolOptions());
+  }
+  
+  /**
+   * Create a connection pool to the database configured with the given {@code connectOptions} and {@code poolOptions}.
+   *
```

Comment:
we need to document that the implementation will look for a pool implementation available in the classpath, return the most specific one or the first one and fail iwhen no implementation is found.